api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -628,16 +632,12 @@ public DbUser syncComplianceTrainingStatusV2(DbUser dbUser, Agent agent)
           BADGE_BY_COMPLIANCE_MODULE.keySet().stream()
               .collect(Collectors.toMap(Function.identity(), determineCompletionTime));
 
-      return updateUserWithRetries(
-          u -> {
-            completionTimes.forEach(
-                (accessModuleName, timestamp) ->
-                    accessModuleService.updateCompletionTime(
-                        u, accessModuleName, timestamp.orElse(null)));
-            return u;
-          },
-          dbUser,
-          agent);
+      completionTimes.forEach(
+          (accessModuleName, timestamp) ->
+              accessModuleService.updateCompletionTime(
+                  dbUser, accessModuleName, timestamp.orElse(null)));
+      updateUserAccessTiers(dbUser, agent);
```

Comment:
For consistency with the previous approach, this would probably return `DbUser`, and that value would be returned here. This may matter if the caller attempts to look at the return value of db user and expects it to reflect the current state - as returned here it could be stale.

Reply:
thanks, updated